proversity_reports_script/google_apis/sheets_api.py

```diff
@@ -0,0 +1,167 @@
+        print('Unable to build or obtain the Google Sheets service. {}'.format(error.message))
+        return None
+
+    return sheet_api
+
+
+def get_settings_from_file(file_path):
+    """
+    Retrives the oAuth credentials dict from a file.
+
+    Args:
+        file_path: oAuth credentials file path.
+    Returns:
+        credentials json object from file.
+        None: if the data cannot be parsed into json format.
+    """
+    credentials = {}
+
+    if os.path.exists(file_path):
+        with open(file_path, 'r') as credentials_file:
+            try:
+                credentials = json.loads(credentials_file.read())
+            except ValueError:
+                print('oAuth config file contents cannot be parsed into json.')
+                return None
+
+    return credentials
+
+
+def get_data_from_csv(file_path):
+    """
+    Reads the csv and returns its data as a list.
+
+    Args:
+        file_path: csv file path.
+    Returns:
+        List containing all rows in the csv file.
+    """
+    csv_data = []
+
+    if os.path.exists(file_path):
+        with open(file_path, 'r') as csvfile:
+            spamreader = csv.reader(csvfile, delimiter=',', quotechar='|')
+
+            for row in spamreader:
+                csv_data.append(row)
+    else:
+        print('The csv file does not exists. {}'.format(file_path))
+    return csv_data
+
+
+def update_sheets_data(file_path, spreadsheet_id):
+    """
+    Updates the report data on the provided spreadsheet id.
+
+    Google Sheets reference: https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values
+
+    Args:
+        file_path: Report file data path.
+        spreadsheet_id: Google Sheet report ID.
+    Returns:
+        None: if there is a problem updating the report.
+    """
+    csv_data = get_data_from_csv(file_path)
+
+    if not csv_data:
+        print('The report data is empty and it was not updated on Google Sheets.')
+        return None
+
+    if not spreadsheet_id:
+        print('Spreadsheet id was not provided and the report cannot be updated on Google Sheets.')
+        return None
+
+    # Updates all the spreadsheet.
+    range_name = 'Sheet1'
+    value_input_option = 'USER_ENTERED'
+    body = {
+        'values': csv_data
+    }
+    api_service = get_sheets_api_service()
+
+    if not api_service:
+        print('Unable to obtain the Google Sheets API service, the report was not updated.')
+        return None
+
+    try:
+        api_service.values().clear(
+            spreadsheetId=spreadsheet_id,
+            range=range_name,
+        ).execute()
+
+        api_service.values().update(
+            spreadsheetId=spreadsheet_id,
+            range=range_name,
+            valueInputOption=value_input_option,
+            body=body,
+        ).execute()
+    except Exception as error:
+        print('There was an error updating report on Google Sheet. {}'.format(error.message))
+        return None
```

Comment:
`None` is not necessary

Reply:
https://www.python.org/dev/peps/pep-0008/#programming-recommendations 
Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, **any return statements where no value is returned should explicitly state this as return None**, and an explicit return statement should be present at the end of the function (if reachable).